ksqldb-parser/src/main/java/io/confluent/ksql/parser/DefaultKsqlParser.java

```diff
@@ -130,4 +144,103 @@ private static String getStatementString(final SingleStatementContext singleStat
         singleStatementContext.stop.getStopIndex()
     ));
   }
+
+  //checks if the error is a reserved keyword error by checking the message and offendingSymbol
+  private static boolean isKeywordError(final String message, final Object offendingSymbol) {
+    final String tokenName = ((CommonToken) offendingSymbol).getText().toLowerCase();
+
+    final Pattern keywordPattern = Pattern.compile("extraneous input.*expecting.*");
+    final Matcher m = keywordPattern.matcher(message);
+    return reservedKeywords.contains(tokenName) && m.find();
+  }
+
+  //Set of keywords minus the non-reserved ones
+  private static final HashSet<String> reservedKeywords = new HashSet<>(Arrays.asList("advance",
```

Comment:
It's better if you create a new class that has a list of the non-reserved keywords and then create there a method that returns all the not-allowed-to-use keywords by using also the `SqlBaseParser.VOCABULARY`. This way, if we ever need to use them in other places, we don't have to hardcode them again. It will make the code reusable and extendable. You can call the class `KsqlParserUtils` and have it public.

Reply:
added